Sources/Protocols/AudioControllerDelegate.swift

```diff
@@ -0,0 +1,45 @@
+/*
+ MIT License
+
+ Copyright (c) 2017-2018 MessageKit
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import Foundation
+
+/// A standard protocol representing an audio controller.
+/// Use this protocol to create your own audio controller object to be used by MessageKit.
+/// MessageKit provide a default implmentation for AudioControllerDelegate in BasicAudioController
+public protocol AudioControllerDelegate: AnyObject {
```

Comment:
Does this need to be a separate delegate or can it be part of `MessagesDisplayDelegate`?

Reply:
The `AudioControllerDelegate` contains the mandatory methods which an AudioController must implement, so any one hwo want a custom implmentation should conform with this protocol. That's way i make it a separate protocol.
I was thinking because `BasicAudioController` updates the `AudioMessageCell` UI it should be part of MessageKit. But it can be moved outside the framework also. If you have any ideea of how the arhitecture for audio can be improved please let me know.